src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -442,7 +444,7 @@ private float convertRowToArrow(
     float rowBufferSize = 0F;
     for (Map.Entry<String, Object> entry : row.entrySet()) {
       rowBufferSize += 0.125; // 1/8 for null value bitmap
-      String columnName = formatColumnName(entry.getKey());
+      String columnName = LiteralQuoteUtils.unquoteColumnName(entry.getKey());
```

Comment:
I don't think this is correct since we need the display name for KC to do schema evolution, for example if the input has a column called "\"c1\"" and the table doesn't have it, then we will return this extra column name in InsertValidationResponse.InsertError and KC will create the column "c1" in the table, so the wrong column will be created if you pass the unquoted name. I believe the unquotedName is only used in the Arrow scanner and all other places should be using the display name, free feel to schedule a chat with me if needed

Reply:
Maybe it is easier to do this the other way around: to use display name only for `InsertValidationResponse.InsertError`
if we send both display and unquoted name from server side?

The problem with using display name to match the user input is that we have to unquote it and remove quote and space escapes to match the server side logic and then quote it back to match against display name. and we still need it for Arrow atm.
We could write custom logic to resolve user input name against display name but then this is yet another way to do things that we do not have on server side but have to keep align.
Now we can just move `unquoteColumnName` to SF common later and reuse it in the SDK.

Reply:
@sfc-gh-tzhang We can discuss it in the evening, but I don't know how support this KC feature without introducing client-side quoting again. For example, if it tries to ingest into column `CREATE` and such column does not exists, KC would have to create it, but it needs to know what it is keyword and therefore quote it, right? Or is the user required to pass it quoted already?